src/core/read_halos-velociraptor.c

```diff
@@ -531,15 +243,21 @@ void read_trees__velociraptor_aug(int snapshot,
 
       double hubble_h = run_globals.params.Hubble_h;
       for (int ii = 0; ii < n_to_read; ii++) {
+        tree_entries[ii].Mass_200crit *= hubble_h * mass_unit_to_internal;
+        tree_entries[ii].Mass_FOF *= hubble_h * mass_unit_to_internal;
+        tree_entries[ii].Mass_tot *= hubble_h * mass_unit_to_internal;
+        tree_entries[ii].R_200crit *= hubble_h;
         tree_entries[ii].Xc *= hubble_h / scale_factor;
         tree_entries[ii].Yc *= hubble_h / scale_factor;
         tree_entries[ii].Zc *= hubble_h / scale_factor;
         tree_entries[ii].VXc /= scale_factor;
         tree_entries[ii].VYc /= scale_factor;
         tree_entries[ii].VZc /= scale_factor;
-        tree_entries[ii].AngMom *= hubble_h * hubble_h * mass_unit_to_internal;
-
-        // TEMPORARY HACK
+        tree_entries[ii].AngMom *= hubble_h * mass_unit_to_internal;
```

Comment:
It's worth double checking. but AngMom is supposed to be (Lx**2+Ly**2+Lz**2)**0.5 / Mvir / Vvir / Rvir / 2**0.5. What you've described get rid of both a hubble_h and mass_unit_initernal. Then if Vvir is calculated through (GMvir/Rvir)**0.5, then there is a mass_unit_to_internal**-0.5, and Rvir has a hubble_h. So overal AngMom needs *= mass_unit_to_internal**-0.5 * hubble_h**-1.

However, it could very likely be that he substutited units and h in Mvir, and then use that to calculate Rvir and Vvir. In this case, units of AngMom simply follows Lx, which needs *= hubble_h**2 * mass_units_to_internal.

Reply:
I am not fully following you here... Before in the non aug trees where AngMom was not stored, we computed it using:
(Lx2+Ly2+Lz2)^0.5 / Mvir (Mvir = Mtot in the merger tree file) after all these quantities were corrected for little_h

While in the aug tree where AngMom is already stored in the tree file, the conversion takes *hubble_h^2 which I don't understand why.
If when Balu computed the trees he didn't substitute any little_h it should be *hubble_h (L goes as hubble_h^2 while M goes as hubble_h so the correction should be *hubble_h)

The factor "mass_unit_to internal" instead makes sense to me as both L and M goes as mass_unit_to_internal so that should cancel out